File: CS/Q296004/Form1.cs
using System;
using DXSample;
using DevExpress.Xpo;
using DevExpress.XtraGrid;
using System.Windows.Forms;
using System.Collections.Generic;
using DevExpress.XtraGrid.Columns;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraEditors.Controls;

namespace Q296004 {
    public partial class Form1 :Form {
        public Form1() {
            InitializeComponent();
        }

        private void gridView1_ValidatingEditor(object sender, BaseContainerValidateEditorEventArgs e) {
            GridView view = (GridView)sender;
            XPBaseObject row = (XPBaseObject)view.GetFocusedRow();
            Dictionary<string, object> customValues = new Dictionary<string, object>();
            customValues.Add(view.FocusedColumn.FieldName, e.Value);
            e.ErrorText = new Validator(row, customValues).Validate(view.FocusedColumn.FieldName);
            e.Valid = string.IsNullOrEmpty(e.ErrorText);
        }

        private void gridView1_ValidateRow(object sender, ValidateRowEventArgs e) {
            if (e.RowHandle != GridControl.NewItemRowHandle) return;
            Dictionary<string, object> customValues = new Dictionary<string, object>();
            GridView view = (GridView)sender;
            foreach (GridColumn col in view.Columns)
                customValues.Add(col.FieldName, view.GetRowCellValue(e.RowHandle, col));
            Validator validator = new Validator((XPBaseObject)e.Row, customValues);
            foreach (GridColumn col in ((GridView)sender).Columns) {
                e.ErrorText = validator.Validate(col.FieldName);
                if (!string.IsNullOrEmpty(e.ErrorText)) {
                    e.Valid = false;
                    return;
                }
            }
        }

        private void Form1_FormClosing(object sender, FormClosingEventArgs e) {
            unitOfWork1.CommitChanges();
        }
    }
}

File: VB/Q296004/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports DXSample
Imports DevExpress.Xpo
Imports DevExpress.XtraGrid
Imports System.Windows.Forms
Imports System.Collections.Generic
Imports DevExpress.XtraGrid.Columns
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraEditors.Controls

Namespace Q296004
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub gridView1_ValidatingEditor(ByVal sender As Object, ByVal e As BaseContainerValidateEditorEventArgs) Handles gridView1.ValidatingEditor
			Dim view As GridView = CType(sender, GridView)
			Dim row As XPBaseObject = CType(view.GetFocusedRow(), XPBaseObject)
			Dim customValues As New Dictionary(Of String, Object)()
			customValues.Add(view.FocusedColumn.FieldName, e.Value)
			e.ErrorText = New Validator(row, customValues).Validate(view.FocusedColumn.FieldName)
			e.Valid = String.IsNullOrEmpty(e.ErrorText)
		End Sub

		Private Sub gridView1_ValidateRow(ByVal sender As Object, ByVal e As ValidateRowEventArgs) Handles gridView1.ValidateRow
			If e.RowHandle <> GridControl.NewItemRowHandle Then
				Return
			End If
			Dim customValues As New Dictionary(Of String, Object)()
			Dim view As GridView = CType(sender, GridView)
			For Each col As GridColumn In view.Columns
				customValues.Add(col.FieldName, view.GetRowCellValue(e.RowHandle, col))
			Next col
			Dim validator As New Validator(CType(e.Row, XPBaseObject), customValues)
			For Each col As GridColumn In (CType(sender, GridView)).Columns
				e.ErrorText = validator.Validate(col.FieldName)
				If (Not String.IsNullOrEmpty(e.ErrorText)) Then
					e.Valid = False
					Return
				End If
			Next col
		End Sub

		Private Sub Form1_FormClosing(ByVal sender As Object, ByVal e As FormClosingEventArgs) Handles MyBase.FormClosing
			unitOfWork1.CommitChanges()
		End Sub
	End Class
End Namespace